src/herder/HerderSCPDriver.cpp

```diff
@@ -687,59 +691,34 @@ HerderSCPDriver::combineCandidates(uint64_t slotIndex,
     }
 
     // take the txSet with the biggest size, highest xored hash that we have
-    TxSetFramePtr bestTxSet;
+    using HighestPair = std::pair<StellarValue, TxSetFramePtr>;
+    HighestPair highestPair = std::accumulate(
+        candidateValues.begin(), candidateValues.end(), HighestPair(),
+        [&](HighestPair const& hp, StellarValue const& sv) {
+            auto const& highestTxSet = hp.second;
+            auto const& svSetHash = sv.txSetHash;
+            auto const cTxSet = mPendingEnvelopes.getTxSet(svSetHash);
+            return (cTxSet && cTxSet->previousLedgerHash() == lcl.hash &&
```

Comment:
The expression here though is quite hard to read (the old code was written like that as to make it straightforward to understand), so maybe preserving the old format helps here?

Reply:
Alright, I removed the factoring-out of the repeatedly-used dereferences (as a result of removing the `std::accumulate` altogether; see below).

I kept the `if` statement squashed, though.  I don't see the nested `if`s as easier to read than one `if` with an `&&`.